controllers/pod.go

```diff
@@ -713,3 +728,272 @@ func getPodNames(pods []*corev1.Pod) []string {
+		found := false
+		for _, statusNamespace := range rackStatus.AerospikeConfig.Value["namespaces"].([]interface{}) {
+			if specNamespace.(map[string]interface{})["name"] == statusNamespace.(map[string]interface{})["name"] {
+				r.Log.Info(
+					"namespace exists",
+				)
+				found = true
+				specStorage := specNamespace.(map[string]interface{})["storage-engine"].(map[string]interface{})
+				statusStorage := statusNamespace.(map[string]interface{})["storage-engine"].(map[string]interface{})
+
+				if specStorage["devices"] != nil {
+					for _, specDevice := range specStorage["devices"].([]interface{}) {
+						deviceFound := false
+						if statusStorage["devices"] != nil {
+							for _, statusDevice := range statusStorage["devices"].([]interface{}) {
+								if specDevice.(string) == statusDevice.(string) {
+									r.Log.Info(
+										"device exist in namespace",
+									)
+									deviceFound = true
+								}
+							}
+						}
+
+						if deviceFound == false {
+							r.Log.Info(
+								"device has been added in namespace",
+							)
+							deviceName := getDeviceNameFromPath(rackState.Rack.Storage.Volumes, specDevice.(string))
+							for key, podStatus := range newAeroCluster.Status.Pods {
+								if podName == key {
+									r.Log.Info(
+										"checking dirty volumes", "device", deviceName, "podname", podName,
+									)
+									if utils.ContainsString(podStatus.DirtyVolumes, deviceName) {
+										return PodRestart, nil
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		if found == false {
+			r.Log.Info(
+				"namespace added",
+			)
+			specStorage := specNamespace.(map[string]interface{})["storage-engine"].(map[string]interface{})
+			if specStorage["type"] == "device" {
+				if specStorage["devices"] != nil {
+					for _, specDevice := range specStorage["devices"].([]interface{}) {
+						deviceName := getDeviceNameFromPath(rackState.Rack.Storage.Volumes, specDevice.(string))
+						for key, podStatus := range newAeroCluster.Status.Pods {
+							if podName == key {
+								r.Log.Info(
+									"checking dirty volumes list", "device", deviceName, "podname", podName,
+								)
+								if utils.ContainsString(podStatus.DirtyVolumes, deviceName) {
+									return PodRestart, nil
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	return QuickRestart, nil
+}
+
+func getDeviceNameFromPath(volumes []asdbv1beta1.VolumeSpec, s string) string {
+	for _, volume := range volumes {
+		if volume.Aerospike.Path == s {
+			return volume.Name
+		}
+	}
+	return ""
+}
+
+func (r *SingleClusterReconciler) deleteFileStorage(pod *corev1.Pod, fileName string) error {
+	r.Log.Info(
+		"file is getting removed", "file", fileName,
+	)
+	cmd := []string{
+		"bash", "-c", fmt.Sprintf(
+			"rm %s",
+			fileName,
+		),
+	}
+
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: r.aeroCluster.Namespace}, pod)
```

Comment:
Any reason for fetching the latest pod and not using the passed pod param ?
The Exec util only cares about pod name/namespace not the actual object